### PR TITLE
Add backups for nextcloud

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

This adds backups to nextcloud. It sets the Nextcloud maintenance mode while doing the backup. The whole /var/www folder will be saved. It also ensures that the database backup is triggered at the same time as well to ensure that both backups are as close together as possible.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: pkg/comp-functions/functions/vshnnextcloud/files/backup.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# We need to run the occ command as the www-data user, but there's no sudo by default
+apt update 1>&2 && apt install sudo -y 1>&2
+
+function disableMaintenance {
+  >&2 echo "Disabling maintenance"
+  sudo -u www-data /var/www/html/occ maintenance:mode --off 1>&2
+}
+
+trap disableMaintenance EXIT
```
Comment (zugao) [inline pkg/comp-functions/functions/vshnnextcloud/files/backup.sh:13]: I just learned what this `trap` is about

File: pkg/comp-functions/functions/vshnnextcloud/deploy.go [hunk 2]
```diff
@@ -136,72 +131,6 @@ func DeployNextcloud(ctx context.Context, svc *runtime.ServiceRuntime) *xfnproto
 	return nil
 }
 
-func addPostgreSQL(svc *runtime.ServiceRuntime, comp *vshnv1.VSHNNextcloud) error {
```
Comment (zugao) [inline pkg/comp-functions/functions/vshnnextcloud/deploy.go:139]: Why are we removing this?

Comment (Kidswiss) [reply pkg/comp-functions/functions/vshnnextcloud/deploy.go:139]: it's dead code. vshnnextcloud uses `common.NewPostgreSQLDependencyBuilder()` that you implemented according to blame :D

File: pkg/comp-functions/functions/vshnnextcloud/files/backup.sh (hunk 1 above)
Comment (Kidswiss) [reply pkg/comp-functions/functions/vshnnextcloud/files/backup.sh:13]: > A day without learning is a day wasted. There is so much to learn and so little time to learn it.

:D

File: pkg/comp-functions/functions/vshnnextcloud/deploy.go (hunk 2 above)
Comment (zugao) [reply pkg/comp-functions/functions/vshnnextcloud/deploy.go:139]: perfect then you fixed my bug, thank you :D

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**